### ISSUE TITLE
Schema generator add illegal request key under requestBody

### ISSUE THREAD

**Issue by nbb-rhajdari** [zircote/swagger-php#1639]
We have a project that relies on nelmio doc bundle which in turn uses swagger-php. When we access our openapi doc.json, we suddenly receive a `request` key, that we did not have before, eg:

```
"requestBody": {
    "request": "ImageController",
    "description": "Replace product images.",
    "required": true,
    "content": {
        "application/json": {
            "schema": {
                "$ref": "#/components/schemas/ReplaceImageRequestDTO"
            }
        }
    }
},
```

As per the spec this key is illegal: https://spec.openapis.org/oas/latest.html#request-body-object

Field Name | Type | Description
-- | -- | --
description | string | A brief description of the request body. This could contain examples of use. [CommonMark] syntax MAY be used for rich text representation.
content | Map[string, Media Type Object] | REQUIRED. The content of the request body. The key is a media type or media type range, see [RFC7231] Appendix D, and the value describes it. For requests that match multiple keys, only the most specific key is applicable. e.g. text/plain overrides text/*
required | boolean | Determines if the request body is required in the request. Defaults to false.

This obviously leads to parsing errors, since the key is illegal:

```
Structural error at paths./api/images.put.requestBody
should NOT have additional properties
additionalProperty: request
Jump to line 438
```

Apparently this change comes from this PR: https://github.com/zircote/swagger-php/pull/1544

So am I right in the assumption that this is a bug or did I miss something?

**Comment by DerManoMann** [comment]
That would be a bug. However, hard to tell in conjunction with the bundle...

I've added some more variations to the request body scratch test https://github.com/zircote/swagger-php/pull/1640 The generated yaml looks ok to me; could you check if your attribute usage is covered by the test and if not post a sample so I can add it?

**Comment by DerManoMann** [comment]
The annotations are written in a way that makes sense when using the library. The `request` property holds the name/key the request body is stored under in `components.requestBodies`.

If you look at the `RequestBody` annotation you'll find that the `request`property is excluded from serialization:

```php
    #[\ReturnTypeWillChange]
    public function jsonSerialize()
    {
        $data = parent::jsonSerialize();

        unset($data->request);

        return $data;
    }
```

This is implemented as `\JsonSerializable`, so I would expect this to apply to nested annotations too. First time I have heard about the serialization not working as expected.

is there any difference in calling `$openapi->toJson()` instead?

**Comment by DerManoMann** [comment]
Interestingly enough, this seems to be a duplicate of #1556 which was resolved via #1557.

Are you using the latest swagger-php version?

**Comment by nbb-rhajdari** [comment]
Sorry for the late reply, I have been absent for a while. You are absolutely right. We had been locked to 4.8.5, just one patch higher and we would have had the fix. Sorry for the confusion. I will close this issue :-)
